lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -265,19 +282,17 @@ newTransactionLayer networkId = TransactionLayer
     _mkDelegationJoinTx poolId acc@(accXPrv, pwd') keyFrom tip cs = do
         let accXPub = toXPub accXPrv
         let certs =
-                if deposit cs > 0 then
-                    [ toStakeKeyRegCert  accXPub
-                    , toStakePoolDlgCert accXPub poolId
-                    ]
-                else
-                    [ toStakePoolDlgCert accXPub poolId ]
+                if deposit cs > 0
+                then mkDelegationCertificates (RegisterKeyAndJoin poolId) accXPub
+                else mkDelegationCertificates (Join poolId) accXPub
 
         let mkWits unsigned =
                 [ mkShelleyWitness unsigned (accXPrv, pwd')
                 ]
 
         let payload = TxPayload Nothing certs mkWits
-        mkTx networkId payload tip acc keyFrom cs
+        let ttl = defaultTTL tip
+        mkTx networkId payload ttl acc keyFrom cs
```

Comment:
Is this the result of a wrong merge-conflict resolution? I don't why this PR would have to change anything regarding transaction TTL?